src/d3d9/d3d9_volume.cpp

```diff
@@ -97,12 +97,43 @@ namespace dxvk {
     if (unlikely(pLockedBox == nullptr))
       return D3DERR_INVALIDCALL;
 
-    return m_parent->LockImage(
+    if (m_texture->Device()->IsD3D8Compatible()) {
+      // D3D8 LockBox clears any existing content present in pLockedBox
+      pLockedBox->pBits = nullptr;
+      pLockedBox->RowPitch = 0;
+      pLockedBox->SlicePitch = 0;
+    }
+
+    if (unlikely(pBox != nullptr)) {
+      auto& desc = *(m_texture->Desc());
+
+      // Negative or zero length dimensions
+      if ( static_cast<LONG>(pBox->Right)  - static_cast<LONG>(pBox->Left)  <= 0
+        || static_cast<LONG>(pBox->Bottom) - static_cast<LONG>(pBox->Top)   <= 0
+        || static_cast<LONG>(pBox->Back)   - static_cast<LONG>(pBox->Front) <= 0
+      // Exceeding surface dimensions
+        || pBox->Right  > desc.Width
+        || pBox->Bottom > desc.Height
+        || pBox->Back   > desc.Depth)
+        return D3DERR_INVALIDCALL;
+    }
+
+    D3DLOCKED_BOX lockedBox;
```

Comment:
It's more of a "defensive coding" thing, since I haven't looked at what can happen on the backend. But ideally with both d3d8 and 9 the expectation is to keep either a cleared box (d3d8) or the initial box values (d3d9) on error. To do that, I mirrored what was already happening in LockRect and copied the returned values at the last possible moment.